### PR TITLE
[OK] Google Contacts Import, AddressBook backend Contact App

### PR DESCRIPTION
**Pull request by monsieurh**
**Purpose**
- Add a program to convert google exported contacts to an agnostic format.
- Presents a basic architecture for storing Contacts in ZPUI

**Implementation**
- python package `vobject` is used to parse vcard format (may be compatible with more than just google, untested)
- `Contact` class contains all data for a contact. All attributes are lists to stay compatible with vcard standard and allow future exports. `Contact,merge(self, other)` and `Contact.match(other)` are worth noting. They allow repectively to merge two contacts into one and to compare how similar they are.
- `AddressBook` is a python `Singleton` that loads, saves and organizes contacts. `AddressBook.search` and `AddressBook.find_duplicates` may be used in external applications

**Discussion**
_The storage format is a `pickle.dump` of the list of contacts_
1. the save file is stored in `$XDG_DATA_HOME`. If not available `~/.zpui/` is chosen (Thanks to @bjoernfan on irc for the suggestion). Do we keep it that way ?
2. The save file is unencrypted and dead-easy to read, what should we do about it ?

### CODE REVIEW

File: apps/personal/contacts/main.py [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+import argparse
+import doctest
+import logging
+import pickle
+import os
+import vobject
+
+from ui.utils import flatten
+
+ZPUI_HOME = "~/.zpui/"  # todo : XDG_DATA_DIR/zpui
```
Comment (CRImier) [inline apps/personal/contacts/main.py]: The contacts app is going to be independent of ZPUI at some point, maybe use `.phone`?

File: ui/utils.py [hunk 2]
```diff
@@ -5,6 +5,14 @@
 logger.setLevel(logging.INFO)
 
 
+def flatten(foo):
```
[the remark this answers is not in the document]
Comment (CRImier) [reply ui/utils.py]: Also, a general-purpose `flatten` helper would be useful - I recently had to reimplement it in the 2048 game, for instance =)

File: apps/personal/contacts/main.py (hunk 1 above)
Comment (monsieurh) [reply apps/personal/contacts/main.py]: this is now the default

**Comment by monsieurh** [comment]
Changes implemented 
- default fallback save folder is now `~/.phone`
- split `AddressBook` and `Contact` class in separate files
- moves `flatter()` in `helpers.general`
- adds an application to browse contact (`personal/contacts`)

**Comment by CRImier** [comment]
Will merge once I wake up =)

**Pull request merged**